docs/API.md

```diff
@@ -1,1611 +1,1611 @@
+
+The charm or bundle is verified before being made available.
+
+The response holds the full charm/bundle id including the revision number.
+
+```
+        type UploadedId struct {
+                Id string
+        }
+```
+
+Example:
+
+```
+{
+    "Id": "precise/wordpress-24"
+}
+```
+
+`DELETE id/archive`
+
+This deletes the given charm or bundle with the given id. ==Change!== (original: If the id does not mention a specific series or revision, all the series and revisions of the given id are deleted. ) If the ID is not fully specified, the charm series or revisions are not resolved and the charm is not deleted. In order to delete the charm, the ID must include series as well as revisions. In order to delete all versions of the charm, use `/expand-id` and iterate on all elements in the result.
+
+
+### Visual diagram
+`GET id/diagram.svg`
+
+This returns a scalable vector-graphics image representing the entity with the given id. This will return a not-found error for charms.
+
+`GET id/icon.svg`
+
+This returns the SVG image of the charm's icon. This reports a not-found error for bundles. Unlike the `id/archive/icon.svg` where 404 is returned in case an icon does not exist, this endpoint returns the default icon.
+
+`GET id/readme`
+
+This returns the README.
+
+### Stats
+
+`GET stats/counter/key[:key]...?[by=unit]&start=date][&stop=date][&list=1]`
+
+The stats path allows the retrieval of counts of operations in a general way. A statistic is composed of an ordered tuple of keys: `kind:series:name:user`. Operations on the store increment counts associated with a specific tuple, determined by the operation and the charm being operated on.
+
+When querying statistics, it is possible to aggregate statistics by using a "\*" as the last tuple element, standing for all tuples with the given prefix. For example, missing:\* will retrieve the counts for all operations of kind "missing", regardless of the series, name or user.
+
+If the list flag is specified, counts for all next level keys will be listed. For example, a query for `stats/counter/download:*?list=1&by=week` will show all the download counts for each series for each week.
+
+If a date range is specified, the returned counts will be restricted to the given date range. Dates are specified in the form yyyy-mm-dd. If the `by` flag is specified, one count is shown for each unit in the specified period, where unit can be `week` or `day`.
+
+Possible kinds are:
+
+* missing
+* info
+* archive-download
+* archive-delete
+* archive-upload
+* archive-failed-upload
+
+Missing is a count of all charms looked up but not found.
+
+==TBD exactly which operations increment which kinds.==
+
+```
+        []Statistic
+
+        type Statistic struct {
+                Key string      `json:",omitempty"`
+                Date string     `json:",omitempty"`
+                Count int64
+        }
+```
+
+The date is in YYYY-MM-DD format. ==Redundant==
```

Comment:
Could you remove this line or the redundant marker? It was a note that snuck through.